internal/cmd/export.go

```diff
@@ -1078,6 +1084,101 @@ func gatherPatroniInfo(ctx context.Context,
 	return nil
 }
 
+// gatherSystemTime takes a client and buffer and collects system time
+// in each Pod and calculates the delta against client system time.
+func gatherSystemTime(ctx context.Context,
+	clientset *kubernetes.Clientset,
+	config *rest.Config,
+	namespace string,
+	clusterName string,
+	tw *tar.Writer,
+	cmd *cobra.Command,
+) error {
+	writeInfo(cmd, "Collecting system times from containers...")
+	// Get the cluster Pods by label
+	pods, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: "postgres-operator.crunchydata.com/cluster=" + clusterName,
+	})
+	if err != nil {
+		if apierrors.IsForbidden(err) {
+			writeInfo(cmd, err.Error())
+			return nil
+		}
+		return err
+	}
+
+	if len(pods.Items) == 0 {
+		// If we didn't find any resources, skip
+		writeInfo(cmd, "PostgresCluster Pods not found when gathering system time information, skipping")
+		return nil
+	}
+
+	podExec, err := util.NewPodExecutor(config)
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+	for _, pod := range pods.Items {
+		for _, container := range pod.Spec.Containers {
+			// Attempt to exec in and run 'date' command in all available containers,
+			// regardless of state, etc.
+			exec := func(stdin io.Reader, stdout, stderr io.Writer, command ...string,
+			) error {
+				return podExec(namespace, pod.GetName(), container.Name,
+					stdin, stdout, stderr, command...)
+			}
+
+			stdout, stderr, err := Executor(exec).systemTime()
+			if err != nil {
+				// If we get an RBAC error, let the user know. Otherwise, just
+				// try the next container.
+				if apierrors.IsForbidden(err) {
+					writeInfo(cmd, fmt.Sprintf(
+						"Failed to get system time for Container \"%s\" in Pod \"%s\". Error: \"%s\"",
+						container.Name, pod.GetName(), err.Error()))
+				}
+				continue
+			}
+
+			// Get client datetime.
+			clientDateTimeStr := time.Now().UTC().Format(time.UnixDate)
+			// Normalize client time for calculating delta.
+			clientTime, err := time.Parse(time.UnixDate, clientDateTimeStr)
+			if err != nil {
+				return err
+			}
```

Comment:
⛏️ I would do this in the other direction to avoid an error. Hold the time, then format it.

```suggestion
			clientTime := time.Now().UTC()
			clientDateTimeStr := clientTime.Format(time.UnixDate)
```

Reply:
I should have made the comment clearer. The normalization here is to zero out time increments under a second. When you run `date` and then parse, milliseconds will be 00 because they're not included in the output: ```Wed Nov  1 07:07:17 AM EDT 2023```. 

When you run time.Now(), the returned value will include milliseconds and nanoseconds, observable with `time.Now().UTC().UnixMilli()`. If you don't set those values to zero, when you perform the subtraction you'll get sub second remainders that we don't care about... but maybe we don't care about taking the delta anyway.

Reply:
That sounds like [Time.Truncate](https://pkg.go.dev/time#Time.Truncate). To ditch sub-second, I think you can call `t = t.Truncate(time.Second)`.